examples/plot_hmm_sampling_and_decoding.py

```diff
@@ -0,0 +1,115 @@
+"""
+Sampling from and decoding an HMM
+---------------------------------
+
+This script shows how to sample points from a Hidden Markov Model (HMM):
+we use a 4-state model with specified mean and covariance.
+
+The plot show the sequence of observations generated with the transitions
+between them. We can see that, as specified by our transition matrix,
+there are no transition between component 1 and 3.
+
+Then, we decode our model to recover the input parameters.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hmmlearn import hmm
+from sklearn.model_selection import RepeatedKFold
+
+# Prepare parameters for a 4-components HMM
+# Initial population probability
+startprob = np.array([0.6, 0.3, 0.1, 0.0])
+# The transition matrix, note that there are no transitions possible
+# between component 1 and 3
+transmat = np.array([[0.7, 0.2, 0.0, 0.1],
+                     [0.3, 0.5, 0.2, 0.0],
+                     [0.0, 0.3, 0.5, 0.2],
+                     [0.2, 0.0, 0.2, 0.6]])
+# The means of each component
+means = np.array([[0.0, 0.0],
+                  [0.0, 11.0],
+                  [9.0, 10.0],
+                  [11.0, -1.0]])
+# The covariance of each component
+covars = .5 * np.tile(np.identity(2), (4, 1, 1))
+
+# Build an HMM instance and set parameters
+gen_model = hmm.GaussianHMM(n_components=4, covariance_type="full")
+
+# Instead of fitting it from the data, we directly set the estimated
+# parameters, the means and covariance of the components
+gen_model.startprob_ = startprob
+gen_model.transmat_ = transmat
+gen_model.means_ = means
+gen_model.covars_ = covars
+
+# Generate samples
+X, Z = gen_model.sample(500)
+
+# Plot the sampled data
+plt.plot(X[:, 0], X[:, 1], ".-", label="observations", ms=6,
+         mfc="orange", alpha=0.7)
+
+# Indicate the component numbers
+for i, m in enumerate(means):
+    plt.text(m[0], m[1], 'Component %i' % (i + 1),
+             size=17, horizontalalignment='center',
+             bbox=dict(alpha=.7, facecolor='w'))
+plt.legend(loc='best')
+plt.show()
+
+# %%
+# Now, let's ensure we can recover our parameters.
+
+scores = list()
+models = list()
+rkf = RepeatedKFold(n_splits=2, n_repeats=10, random_state=99)
```

Comment:
I am confused: how can you get the correct transition matrix after k-fold splitting the input timeseries?

Reply:
Ah good point, these should be split once. I was just having trouble random seeding the Baum-Welch algorithm so I did a bit of a hack and changed up the training data but I should just seed it.

Reply:
Good point! I just split 50/50. I was doing that because I'm having trouble seeding it: I get the same score every time for the same number of components. The version I just pushed has that issue, do you know how to fix it?

Reply:
Ah, okay, I looked into the code a bit more and the `random_state` is used to seed KMeans clustering for the prior on the means. I think maybe let's just start with this and I can add a more complex example that has different KMeans solutions at different random states and show how to seed the EM algorithm in a followup. Probably good to keep the first example simple.